### PR TITLE
Dcos 602 support wildcards

### PR DESCRIPTION
**Pull request by YuraTolochkevich**

### CODE REVIEW

**Comment by jsancio** [comment]
@ssk2 please look into this.

File: dcos/package.py [hunk 1]
```diff
@@ -483,13 +484,27 @@ def _search_rank(pkg, query):
...
+            q = '^{}'.format(q)
+        else:
+            q = '{}$'.format(q)
 
-    result = 0.0
+        if re.match(q, pkg['name'].lower()):
+            result += 2.0
+        return result
+
+    if pkg['name'].lower().startswith(q):
```
Comment (ssk2) [inline dcos/package.py]: Why change this to `startswith` now?

File: cli/tests/integrations/test_package.py [hunk 2]
```diff
@@ -696,3 +696,30 @@ def _package(name,
...
+    assert b'chronos' in stdout
+    assert b'cassandra' in stdout
+    assert stderr == b''
+
+    registries = json.loads(stdout.decode('utf-8'))
+    for registry in registries:
+        assert len(registry['packages']) == 2
+
+
+def test_search_start_with_wildcard():
```
Comment (ssk2) [inline cli/tests/integrations/test_package.py]: Can we add one more test where the wildcard is in the middle of the package name?

File: dcos/package.py (hunk 1 above)
Comment (YuraTolochkevich) [inline dcos/package.py]: this string is redundant, and can be removed

**Comment by YuraTolochkevich** [comment]
@ssk2 Tanks Sunil! Please see updates

**Comment by ssk2** [comment]
Thanks @YuraTolochkevich

File: cli/tests/integrations/test_package.py [hunk 3]
```diff
@@ -703,3 +703,43 @@ def _package(name,
         yield
     finally:
         assert_command(['dcos', 'package', 'uninstall', name])
+
+
+def test_search_ends_with_wildcard():
```
Comment (mgummelt) [inline cli/tests/integrations/test_package.py]: please add tests *before* the private functions.  private functions should be at the end of the file

**Pull request merged**